### ISSUE TITLE
in-repo-addon blacklisting issues

### ISSUE THREAD

**Issue by Windvis** [embroider-build/embroider#302]
Hey!

I'm trying out Embroider in our project and I encountered some build error which I'm not sure how to resolve.

```
The "path" argument must be of type string. Received type undefined
```
This error is triggered by [this line of code](https://github.com/embroider-build/embroider/blob/7b7be78744150e79de101ea3189898da9ee29688/packages/compat/src/compat-addons.ts#L107), because the name property is undefined, but only for our in-repo-addons (it seems).

I tried debugging to find some clues, and it looks like it has something to do with the [`this.packageCache.moved`](https://github.com/embroider-build/embroider/blob/7b7be78744150e79de101ea3189898da9ee29688/packages/compat/src/compat-addons.ts#L103) map. The name property is defined in the key object, but not in the value. I tried following the breadcrumbs through the codebase to see where this `moved` map is created, but I feel it's way out of my league, and it's getting late as well :sleepy: :smile: .

Any pointers about what the problem could be?

**Comment by lifeart** [comment]
Hi @Windvis! Thank you for report! Could you please, create minimal repoduction repo?

**Comment by Windvis** [comment]
Hmm, I just tried setting up a new ember app + in-repo-addon + Embroider, and it builds perfectly fine.. So the in-repo-addon error might be a symptom of a different problem. Any pointers how to debug this further?

**Comment by ef4** [comment]
Do your in-repo addons have names? They need a valid package.json with a name. The generator gives them one, which may be why your new reproduction didn't fail.

**Comment by Windvis** [comment]
Yea, they do. That's one of the first things I checked as well. When I debugged the code I also saw that they all had names. It just seems to get lost somewhere when "moving" them. The keys in the `packageCache.moved` map contain the name, but they are undefined in the value objects. I couldn't find where that map was being built. Anyway, since the repro project didn't have any problems I guess it's something we do wrong in our project. It's just hard to find out what :smile:.

**Comment by Windvis** [comment]
@ef4 I did some more digging and I found out that it is related to the blacklisting of in-repo-addons.

We created an in-repo-addon wrapper around bugsnag which we blacklist during development. Once I removed the blacklist everything compiled as usual. (I didn't test if this also affects blacklisting of normal addons).

I created a [minimal reproduction repo](https://github.com/Windvis/embroider-blacklist-example-repo) for the in-repo-addon blacklist scenario. Building the master branch will fail with the aforementioned error, while building on the `pre-blacklist` branch will work.
